wp-content/themes/borderzine/functions.php

```diff
@@ -101,4 +101,80 @@ function borderzine_has_avatar( $email ) {
 		}
 	}
 	return false;
+}
+
+/**
+ * Modifies the default WordPress search query to see if the 
+ * search terms match any author names and return any posts found by said author
+ * 
+ * @param String $posts_search SQL query for the search
+ * @param Object $wp_query_obj The current wp_query object
+ * @return String $posts_search SQL query for the search
+ * @since 0.1
+ */
+function borderzine_search_posts_by_author( $posts_search, $wp_query_obj ) {
+
+	if ( !is_search() || empty( $posts_search ) ) {
+		return $posts_search;
+	}
+
+	global $wpdb;
+
+	// search all authors to see if any having a name that matches the search term
+	$search = sanitize_text_field( get_query_var( 's' ) );
+	$args = array(
+		'count_total' => false,
+		'search' => sprintf( '*%s*', $search ),
+		'fields' => 'ID',
+		'has_published_posts' => true,
```

Comment:
Well, it looks like this doesn't quite work as described. I'm still seeing INN Labs as an author at https://borderzine.test/?s=INN+Labs&search+submit= . 🤷‍♂

Reply:
Ah, probably because this is using `get_users` instead of `WP_User_Query`

Reply:
@benlk Could also be because the `INN Labs` user has created attachments/pop-ups which are still considered posts. Searching the `wp_posts` table for posts with `post_author = 13704` (INN Labs ID) returns 50 results for my local environment.

Reply:
However, I created a new user with 0 posts, attachments, pop-ups, etc. and searched for it and it didn't return their author page.

Reply:
That would explain it. And setting this line to `'has_published_posts' => array( 'post' ),` doesn't work because post 35602 has type "post", though it is trashed.